src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+            text.Slice(0, length).CopyTo(_byteMap.AsSpan(_position));
+            Refresh();
+        }
+
+        /// <summary>
+        /// Moves the cursor to an explicit column and row position.
+        /// </summary>
+        /// <param name="left">The column position from left to right starting with 0 to 83.</param>
+        /// <param name="top">The row position from the top starting with 0 to 5.</param>
+        /// <exception cref="ArgumentOutOfRangeException">The given position is not inside the display.</exception>
+        public void SetCursorPosition(int left, int top)
+        {
+            if ((left < 0 || left > Size.Width) || (top < 0 || top > Size.Height))
+            {
+                throw new ArgumentOutOfRangeException($"The given position is not inside the display. it's 6 raws and 84 columns");
+            }
+
+            _position = left + top * Size.Width;
+            SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn : FunctionSet.PowerOff), (byte)((byte)(SetAddress.XAddress) | left), (byte)((byte)(SetAddress.YAddress) | top) });
+        }
+
+        #endregion
+
+        #region Drawing points, lines and rectangles
+
+        /// <summary>
+        /// Draw a point
+        /// </summary>
+        /// <param name="x">The X coordinate.</param>
+        /// <param name="y">The Y coordinate.</param>
+        /// <param name="isOn">True if the point has pixels on, false for off.</param>
+        /// <returns>True if success</returns>
+        public bool DrawPoint(int x, int y, bool isOn)
+        {
+            if (x < 0 || x >= 84 || y < 0 || y >= 48)
+            {
+                return false;
+            }
+
+            int index = ((x % 84) + (int)(y * 0.125) * 84);
+
+            byte bitMask = (byte)(1 << (y % 8));
+
+            if (isOn)
+            {
+                _byteMap[index] |= bitMask;
+            }
+            else
+            {
+                _byteMap[index] &= (byte)~bitMask;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Draw a point
+        /// </summary>
+        /// <param name="point">The point to draw.</param>
+        /// <param name="isOn">True if the point has pixels on, false for off.</param>
+        /// <returns></returns>
+        public bool DrawPoint(Point point, bool isOn) => DrawPoint(point.X, point.Y, isOn);
+
+        /// <summary>
+        /// Draw a line
+        /// </summary>
+        /// <param name="x1">The first point X coordinate.</param>
+        /// <param name="y1">The first point Y coordinate.</param>
+        /// <param name="x2">The second point X coordinate.</param>
+        /// <param name="y2">The second point Y coordinate.</param>
+        /// <param name="isOn">True if the line has pixels on, false for off.</param>
+        public void DrawLine(int x1, int y1, int x2, int y2, bool isOn)
```

Comment:
I generally think utilities like this should be on generic abstractions but since we don't have display abstraction I'm fine with that as is for now.